src/sync/Synchroniser.js

```diff
@@ -81,6 +84,39 @@ export class Synchroniser {
     this.batchSize = MIN_SYNC_BATCH_SIZE;
   };
 
+  fetchWrapper = async (url, opts) => {
+    const { body } = opts;
+
+    const bugsnagNotify = (message, response) => {
+      bugsnagClient.notify(new Error(message), {
+        context: 'SYNC ERROR',
+        metaData: { url, body, responseText: response.text() },
+      });
+    };
+    // eslint-disable-next-line no-undef
+    const response = await fetch(url, opts);
+    if (response.status < 200 || response.status >= 300) {
+      throw new Error('Connection failure while attempting to sync.');
+    }
+    let responseJson;
+    try {
+      responseJson = await response.json();
+    } catch (error) {
+      bugsnagNotify('Failed parsing response form server', response);
+      throw new Error('Unexpected response from sync server');
+    }
+
+    if (responseJson.error && responseJson.error !== '') {
+      if (responseJson.error.startsWith("Site registration doesn't match.")) {
+        throw new Error(responseJson.error);
+      } else {
+        bugsnagNotify(responseJson.error, response);
+        throw new Error('Server rejected pushed records');
+      }
+    }
+    return { responseJson };
```

Comment:
maybe returning `responseJson` might be a bit nicer

Reply:
this way we can easily add extra return from fetchWrapper in the future